src/Tank.cpp

```diff
@@ -63,9 +96,32 @@ Tank &Tank::operator=(const Tank &rhs) {
   mTower = rhs.mTower;
   mShot = rhs.mShot;
   mEngine = rhs.mEngine->copy();
+  mTracesHandler =
+      std::make_unique<TracesHandler>(rhs.mTracesHandler->get_trace_texture(), mBody.get_sprite(),
+                                      mPos, rhs.mTracesHandler->get_config());
   return *this;
 }
 
+Tank &Tank::operator=(Tank &&rhs) {
+  if (this == &rhs) {
+    return *this;
+  }
+  mPos = std::move(rhs.mPos);
+  mCurrentSpeed = std::move(rhs.mCurrentSpeed);
+  mShotStart = std::move(rhs.mShotStart);
+  mDrawShot = std::move(rhs.mDrawShot);
+  mBody = std::move(rhs.mBody);
+  mTower = std::move(rhs.mTower);
+  mShot = std::move(rhs.mShot);
+  mEngine = std::move(rhs.mEngine);
+  mTracesHandler =
+      std::make_unique<TracesHandler>(rhs.mTracesHandler->get_trace_texture(), mBody.get_sprite(),
+                                      mPos, rhs.mTracesHandler->get_config());
+  return *this;
+}
+
+Tank::~Tank() = default;
```

Comment:
This should be only in header file

Reply:
I put it in header file at first, but then I received a compilation error, because I use forward declaration for `TracesHandler`. At the moment when compiler generates destructor body full implementation of `TracesHandler` must be already known, otherwise there is a compilation error. It was surprising to me, because I did not have similar problems with `Engine` class.